### PR TITLE
Enable filter on data.content.

### PR DESCRIPTION
**Pull request by perolavsvendsen**
## Issue
#249 

## Short description
`data.content` is a key metadata attribute. Unlike non-standard freetext attributes such as "tagname" and others which should never be used for logical operations, the `data.content` is standardized. Therefore, we want to promote usage of `data.content` as much as possible.

This PR should enable explorer to filter on `data.content`.

## Pre-review checklist
- [x] Read through all changes. No redundant `print()` statements, commented-out code, or other remnants from the development. 👀
- [x] Make sure tests pass after every commit. ✅
- [x] New/refactored code is following same conventions as the rest of the code base. 🧬
- [x] New/refactored code is tested. ⚙
- [x] Documentation has been updated 🧾

## Pre-merge checklist
- [x] Commit history is consistent and clean. 👌

### CODE REVIEW

File: examples/explorer.ipynb [hunk 1]
```diff
@@ -127,22 +133,23 @@
...
     "\n",
     "print(\"\\nSelected surface:\")\n",
-    "print(\"ID:\", surf.uuid)\n",
+    "print(\"Uuid:\", surf.uuid)\n",
     "print(\"Name:\", surf.name)\n",
+    "print(\"Content:\", surf.content)\n",
     "print(\"Tagname:\", surf.tagname)\n",
     "print(\"Iteration:\", surf.iteration)\n",
     "print(\"Realization:\", surf.realization)\n",
-    "print(\"stratigraphic\", surf.stratigraphic)\n",
```
Comment (perolavsvendsen) [inline examples/explorer.ipynb:144]: Removed "stratigraphic" from example, not very relevant for end user.

File: examples/explorer.ipynb [hunk 2]
```diff
@@ -127,22 +133,23 @@
     "# Iterate over results\n",
     "print(\"\\nResults:\", len(surfs))\n",
     "\n",
-    "for surf in surfs:\n",
-    "    print(\"\\n\")\n",
-    "    print(\"ID:\", surf.uuid)\n",
-    "    print(\"Realization:\", surf.realization)\n",
+    "# Sort by realization\n",
```
Comment (perolavsvendsen) [inline examples/explorer.ipynb:136]: Sorted output on `realization` to make it more relatable

File: src/fmu/sumo/explorer/objects/_child.py [hunk 3]
```diff
@@ -23,6 +24,11 @@ def name(self) -> str:
         """Object name"""
         return self._get_property(["data", "name"])
 
+    @property
```
Comment (perolavsvendsen) [inline src/fmu/sumo/explorer/objects/_child.py:27]: Main change in this PR: Add `content` property to the child object.

**Review by rwiker — APPROVED** [review]
LGTM.

File: examples/explorer.ipynb (hunk 2 above)
Comment (perolavsvendsen) [reply examples/explorer.ipynb]: This is now gone, removed due to conflict with current main

File: docs/explorer.rst [hunk 4]
```diff
@@ -243,7 +249,7 @@ The `SurfaceCollection.filter` method takes the following parameters:
 
 * uuid
 * name 
-* tagname
```
Comment (rwiker) [inline docs/explorer.rst:246]: Is it intentional to remove `tagname`?

**Review by equinor-ruaj — APPROVED** [review]
lgtm

**Pull request merged**